sky/admin_policy.py

```diff
@@ -0,0 +1,97 @@
+"""Interface for admin-defined policy for user requests."""
+import abc
+import dataclasses
+import typing
+from typing import Optional
+
+if typing.TYPE_CHECKING:
+    import sky
+
+
+@dataclasses.dataclass
+class RequestOptions:
+    """Request options for admin policy.
+
+    Args:
+        cluster_name: Name of the cluster to create/reuse.
+        cluster_running: Whether the cluster is running.
+        idle_minutes_to_autostop: If provided, the cluster will be set to
+            autostop after this many minutes of idleness.
+        down: Whether to down the cluster.
+        dryrun: Whether to dryrun the request.
+    """
+    cluster_name: Optional[str]
+    cluster_running: bool
+    idle_minutes_to_autostop: Optional[int]
+    down: bool
+    dryrun: bool
+
+
+@dataclasses.dataclass
+class UserRequest:
+    """User request to the policy.
+
+    It is a combination of a task, request options, and the global skypilot
+    config used to run a task, including `sky launch / exec / jobs launch / ..`.
```

Comment:
```suggestion
    """A user request.
    
    A "user request" is defined as a `sky launch / exec / jobs launch / serve up` command or its API equivalent.

    This class wraps the underlying task, the global skypilot
    config used to run a task, and the request options.
```

- Btw, this hasn't explained what happens if the request is a `jobs launch` with a pipeline; what does the singleton `task` represent in that case? 
- Or maybe the definition is incorrect -- should it exclude `jobs launch`? It sounds like a single `jobs launch` may translate into several such "user requests", one per task.

We should update the docstr to reflect the above.

Reply:
Rephrased. PTAL ; )